### PR TITLE
Show all imagestreams in imagestream file

### PR DESCRIPTION
**Pull request by phracek**
This pull request shows all tags in all imagestream json files.

It replaces this pull request https://github.com/sclorg/container-common-scripts/pull/256

### CODE REVIEW

**Comment by phracek** [comment]
In real usecase, I tested it on postgresql-container.

```
python3 /root/postgresql-container/test/show_all_imagestreams.py
Tags in the image stream imagestreams/postgresql-centos.json:
- 10-el8 -> quay.io/sclorg/postgresql-10-c8s:latest
- 12-el8 -> quay.io/sclorg/postgresql-12-c8s:latest
- 13-el8 -> quay.io/sclorg/postgresql-13-c8s:latest
- 15-el8 -> quay.io/sclorg/postgresql-15-c8s:latest
- 13-el9 -> quay.io/sclorg/postgresql-13-c9s:latest
- 15-el9 -> quay.io/sclorg/postgresql-15-c9s:latest
- 16-el9 -> quay.io/sclorg/postgresql-16-c9s:latest
- latest -> 16-el9
Tags in the image stream imagestreams/postgresql-rhel-aarch64.json:
- 10-el8 -> registry.redhat.io/rhel8/postgresql-10:latest
- 12-el8 -> registry.redhat.io/rhel8/postgresql-12:latest
- 13-el8 -> registry.redhat.io/rhel8/postgresql-13:latest
- 15-el8 -> registry.redhat.io/rhel8/postgresql-15:latest
- 13-el9 -> registry.redhat.io/rhel9/postgresql-13:latest
- 15-el9 -> registry.redhat.io/rhel9/postgresql-15:latest
- 16-el9 -> registry.redhat.io/rhel9/postgresql-16:latest
- latest -> 16-el9
Tags in the image stream imagestreams/postgresql-rhel.json:
- 10-el7 -> registry.redhat.io/rhscl/postgresql-10-rhel7:latest
- 12-el7 -> registry.redhat.io/rhscl/postgresql-12-rhel7:latest
- 13-el7 -> registry.redhat.io/rhscl/postgresql-13-rhel7:latest
- 10-el8 -> registry.redhat.io/rhel8/postgresql-10:latest
- 12-el8 -> registry.redhat.io/rhel8/postgresql-12:latest
- 13-el8 -> registry.redhat.io/rhel8/postgresql-13:latest
- 15-el8 -> registry.redhat.io/rhel8/postgresql-15:latest
- 13-el9 -> registry.redhat.io/rhel9/postgresql-13:latest
- 15-el9 -> registry.redhat.io/rhel9/postgresql-15:latest
- 16-el9 -> registry.redhat.io/rhel9/postgresql-16:latest
- latest -> 16-el9
```

File: tests/check_imagestreams.sh [hunk 1]
```diff
@@ -3,7 +3,12 @@
 set -x
 
 check_imagestreams=$(dirname "$(readlink -f "$0")")/../check_imagestreams.py
+show_all_imagestreams=$(dirname "$(readlink -f "$0")")/../show_all_imagestreams.py
 "${PYTHON-python3}" "$check_imagestreams" "2.5"
 test $? -eq 1
 "${PYTHON-python3}" "$check_imagestreams" "2.4"
 test $? -eq 0
+echo "Check full is output"
```
Comment (zmiklank) [inline tests/check_imagestreams.sh:11]: Could you please rephrase this string? I am not sure what it means.

File: test-lib.sh [hunk 2]
```diff
@@ -1147,9 +1148,11 @@ ct_check_latest_imagestreams() {
     [ -f "$latest_version/.exclude-$OS" ] && latest_version=$(grep '^VERSIONS' Makefile | rev | cut -d ' ' -f 2 | rev )
     # Only test the imagestream once, when the version matches
     # ignore the SC warning, $VERSION is always available
+
+    test_lib_dir=$(dirname "$(readlink -f "$0")")
+    python3 "${test_lib_dir}/show_all_imagestreams.py"
```
Comment (zmiklank) [inline test-lib.sh:1153]: Do we want to test something with calling `show_all_imagestreams.py`, or is it here for debugging purposes?

**Review by zmiklank — COMMENTED** [review]
Thank you for the PR. Please see my comments.

File: test-lib.sh (hunk 2 above)
Comment (phracek) [reply test-lib.sh:1153]: Test is present here: https://github.com/sclorg/container-common-scripts/pull/372/files#diff-9c2d3c45e3a4d98424608dc9f1aad1ed04f13cd5d79c3364c6d2a9dd99548dd6R13

Comment (phracek) [reply test-lib.sh:1153]: The output is only for information.

File: tests/check_imagestreams.sh (hunk 1 above)
Comment (phracek) [reply tests/check_imagestreams.sh]: Fixed by https://github.com/sclorg/container-common-scripts/pull/372/commits/1903527ebf5564ecb8ca5b9c155dbeb770a76d08. Good point.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: show_all_imagestreams.py [hunk 3]
```diff
@@ -57,8 +53,9 @@ def show_all_imagestreams(self):
             print(f"Tags in the image stream {f}:")
             for tag in json_dict["spec"]["tags"]:
                 print(f"- {tag['name']} -> {tag['from']['name']}")
+        return 0
 
 
 if __name__ == "__main__":
     isc = ShowAllImageStreams()
-    isc.show_all_imagestreams()
+    sys.exit(isc.show_all_imagestreams())
```
Comment (zmiklank) [inline show_all_imagestreams.py:61]: Why do we need exiting the script with sys.exit?

**Review by zmiklank — COMMENTED** [review]
One more thing, otherwise LGTM, thanks.

File: show_all_imagestreams.py (hunk 3 above)
Comment (phracek) [reply show_all_imagestreams.py:61]: Good point. We do not need it. I have used in script check_imagestreams.py.

Comment (phracek) [reply show_all_imagestreams.py]: Fixed by https://github.com/sclorg/container-common-scripts/pull/372/commits/43d7e4f42d6655ba35492b90cb5dd30dc085d449

**Comment by zmiklank** [comment]
@phracek In that case we can also remove "import sys".

**Pull request merged**